modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -69,6 +69,7 @@ type TItemsActions = {
 	icon?: string;
 	id?: string;
 	label?: string;
+	onClick?: Function;
```

Comment:
So you added `onClick` to `TItemsActions`, what do I do if I need a small modal as a `target`? Should I add the type as a part of my changes?

Reply:
If you need it, yes. We don't need to be comprehensive about types, adding only used properties. It's quite likely that we have unused ones. We can make a comprehensive sweep when we migrate FDS to TS. 

On a sidenote, you should probably not use modal target, as this uses custom FDS modal. We intend to replace it with DXP standard `openModal`. In general, `action.target` is not a great API. The main reason for it is to have configuration in Java. You set a keyword, `target`, and this determines hardcoded frontend behavior. We don't need to limit ourselves in that way when using `FrontendDataSet` React component directly.

Reply:
Okay, so I guess something like an onClick that opens our frontend-js-web modal would be the preferred route?

Reply:
yep :+1: